### PR TITLE
feat: Enable resource status for telemetry resources

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- Remove feature flag `KymaInputAllowed`
- Remove flag from telemetry manager controller
- Allow e2e tests to run on non-dev version
- update unit tests

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/1294

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

File: docs/user/04-metrics.md [hunk 1]
```diff
@@ -347,6 +347,14 @@ spec:
 > [!NOTE]
 > The Metric agent can scrape endpoints even if the workload is a part of the Istio service mesh and accepts mTLS communication. However, there's a constraint: For scraping through HTTPS, Istio must configure the workload using 'STRICT' mTLS mode. Without 'STRICT' mTLS mode, you can set up scraping through HTTP by applying the annotation `prometheus.io/scheme=http`. For related troubleshooting, see [Log entry: Failed to scrape Prometheus endpoint](#log-entry-failed-to-scrape-prometheus-endpoint).
 
+
+### 4. Default Metrics for Telemetry Health
+By default, the metric gateway has the [kymastats receiver](https://github.com/kyma-project/opentelemetry-collector-components/blob/main/receiver/kymastatsreceiver/README.md) enabled. This receiver collects [metrics](https://github.com/kyma-project/opentelemetry-collector-components/blob/main/receiver/kymastatsreceiver/documentation.md) to monitor the health of following resources if they have been deployed:
```
Comment (a-thaler) [inline docs/user/04-metrics.md:352]: Please make it much more end-user oriented and do not talk about implementation details, like:

By default, a MetricPipeline will emit metrics about the health of all pipelines managed by the telemetry module. On base of this metrics, you can track the status of every individual pipeline and setup alerting for it. The following metric is emmitted for every pipeline and the telemetry module itself:
- xxx.conditions - the value represents the xxx

The following metric attributes are available on the metric:
- xxx - use it to narrow down the bla

A typical alert rule will look like that:

Comment (NHingerl) [reply docs/user/04-metrics.md:352]: ;) And language-edited:

By default, a MetricPipeline emits metrics about the health of all pipelines managed by the Telemetry module. Based on these metrics, you can track the status of every individual pipeline and set up alerting for it. The following metric is [or, if there's more than one metric: "The following metrics are..."] emitted for every pipeline and the Telemetry module itself:

xxx.conditions - the value represents the xxx
The following metric attributes are available for the metric:

xxx - use it to narrow down the bla
A typical alert rule looks like the following example:

File: docs/user/04-metrics.md [hunk 2]
```diff
@@ -349,11 +349,27 @@ spec:
...
-- Telemetry
-- Log Pipeline
-- Trace Pipeline
-- Metric Pipeline
+By default, a MetricPipeline emits metrics about the health of all pipelines managed by the Telemetry module. Based on these metrics, you can track the status of every individual pipeline and set up alerting for it. The following metrics are emitted for every pipeline and the Telemetry module itself:
+
+| metric                          | description                                                              | availability                                       |
+|---------------------------------|--------------------------------------------------------------------------|----------------------------------------------------|
+| kyma.resource.status.conditions | value represents status of different conditions reported by the resource | available for both pipeline and telemetry resource |
```
Comment (NHingerl) [inline docs/user/04-metrics.md:356]:
```suggestion
| kyma.resource.status.conditions | Value represents status of different conditions reported by the resource | Available for both, the pipeline and the telemetry resource |
```

Consistent capitalization

File: docs/user/04-metrics.md [hunk 3]
```diff
@@ -349,11 +349,27 @@ spec:
...
+| kyma.resource.status.state      | value represents the state of the resource (if present)                  | available for telemetry resource                   |
+
+The following metric attributes are available for the metric that could be used for monitoring:
```
Comment (NHingerl) [inline docs/user/04-metrics.md:359]: why "could"? Is this uncertain or optional?

File: docs/user/04-metrics.md [hunk 4]
```diff
@@ -349,11 +349,27 @@ spec:
...
-- Telemetry
-- Log Pipeline
-- Trace Pipeline
-- Metric Pipeline
+By default, a MetricPipeline emits metrics about the health of all pipelines managed by the Telemetry module. Based on these metrics, you can track the status of every individual pipeline and set up alerting for it. The following metrics are emitted for every pipeline and the Telemetry module itself:
+
+| metric                          | description                                                              | availability                                       |
+|---------------------------------|--------------------------------------------------------------------------|----------------------------------------------------|
+| kyma.resource.status.conditions | value represents status of different conditions reported by the resource | available for both pipeline and telemetry resource |
+| kyma.resource.status.state      | value represents the state of the resource (if present)                  | available for telemetry resource                   |
```
Comment (NHingerl) [inline docs/user/04-metrics.md:357]:
```suggestion
| kyma.resource.status.state      | Value represents the state of the resource (if present)                  | Available for the telemetry resource                   |
```

File: docs/user/04-metrics.md [hunk 5]
```diff
@@ -349,11 +349,27 @@ spec:
...
+
+The following metric attributes are available for the metric that could be used for monitoring:
+
+| Name                     | names                                                                                               |
+|--------------------------|-----------------------------------------------------------------------------------------------------|
+| metric.attributes.Type   | Type of the condition                                                                               |
+| metric.attributes.status | status of the condition                                                                             |
```
Comment (NHingerl) [inline docs/user/04-metrics.md:364]:
```suggestion
| metric.attributes.status | Status of the condition                                                                             |
```

File: docs/user/04-metrics.md [hunk 6]
```diff
@@ -349,11 +349,27 @@ spec:
...
+| metric.attributes.reason | reason contains a programmatic identifier indicating the reason for the condition's last transition |
```
Comment (NHingerl) [inline docs/user/04-metrics.md:365]:
```suggestion
| metric.attributes.reason | Contains a programmatic identifier indicating the reason for the condition's last transition |
```

File: docs/user/04-metrics.md [hunk 7]
```diff
@@ -349,11 +349,27 @@ spec:
...
+
+
+A typical alert rule for looks like the following example. This alert would be triggered if metrics are not delivered to the backend for 5 minutes:
```
Comment (NHingerl) [inline docs/user/04-metrics.md:368]: - between "for" and "looks", a word is missing. If there's no word to add, remove the "for".
- Why "would be", why not "is"?

```suggestion
A typical alert rule for [ADD WORD] looks like the following example. The alert is triggered if metrics are not delivered to the backend for 5 minutes:
```

**Review by hisarbalik — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**